src/radical/pilot/task_manager.py

```diff
@@ -830,52 +834,157 @@ def submit_units(self, descriptions):
+    def submit_raptors(self, descriptions, pilot_id=None):
+        '''
+        Submits on or more :class:`radical.pilot.TaskDescription` instances to
+        the task manager, where the `TaskDescriptions` have the mode
+        `RAPTOR_MASTER` set.
+
+        This is a thin wrapper around `submit_tasks`.
+
+        **Returns:**
+              * A list of :class:`radical.pilot.Raptor` objects.
+        '''
+
+        descriptions = ru.as_list(descriptions)
+
+        for td in descriptions:
+
+            td.pilot = pilot_id
+            td.mode  = RAPTOR_MASTER
+
+            raptor_file   = td.get('raptor_file')  or  ''
+            raptor_class  = td.get('raptor_class') or  'Master'
+
+            td.arguments  = [raptor_file, raptor_class]
+
+            td.environment['PYTHONUNBUFFERED'] = '1'
+
+            if not td.get('uid'):
+                td.uid = ru.generate_id('raptor.%(item_counter)04d',
+                                        ru.ID_CUSTOM, ns=self._session.uid)
+
+            if not td.get('executable'):
+                td.executable = 'radical-pilot-raptor-master'
+
+            if not td.get('named_env'):
+                td.named_env = 'rp'
+
+        return self.submit_tasks(descriptions)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def submit_workers(self, descriptions):
         """
+        Submits on or more :class:`radical.pilot.TaskDescription` instances to
+        the task manager, where the `TaskDescriptions` have the mode
+        `RAPTOR_WORKER` set.
+
+        uid:  (str) ID of raptor master this worker will be associated with
+
+        This is a thin wrapper around `submit_tasks`.
+
+        **Returns:**
+              * A list of :class:`radical.pilot.RaptorWorker` objects.
+        """
+
+        descriptions = ru.as_list(descriptions)
+
+        for td in descriptions:
+
+            raptor_id    = td.get('raptor_id')
+            raptor_file  = td.get('raptor_file')  or  ''
+            raptor_class = td.get('raptor_class') or  'DefaultWorker'
+
+            if not raptor_id:
+                raise ValueError('RAPTOR_WORKER descriptions need `raptor_id`')
+
+            if not td.get('uid'):
+                td.uid = ru.generate_id(raptor_id + '.%(item_counter)04d',
+                                        ru.ID_CUSTOM, ns=self._session.uid)
+
+            if not td.get('executable'):
+                td.executable = 'radical-pilot-raptor-worker'
+
+            if not td.get('named_env'):
+                td.named_env = 'rp'
+
+            td.mode      = RAPTOR_WORKER
+            td.raptor_id = raptor_id
+            td.arguments = [raptor_file, raptor_class, raptor_id]
+
+            td.environment['PYTHONUNBUFFERED'] = '1'
+
+        return self.submit_tasks(descriptions)
+
+
+    # --------------------------------------------------------------------------
+    #
+    def submit_tasks(self, descriptions=None, tasks=None):
```

Comment:
have a general proposal here - to have just tasks (`submit_tasks(self, tasks=None)`), and later while going through them (since we still go through all of them) check if certain item is `TD` or `Task`. With the quick look on where `submit_tasks` is called (and in docstrings), need to ensure (or clear up) whether task descriptions are provided or tasks themselves.

Reply:
I agree with that but differently: I reverted the change.  The move to the changed task submission should be a separate PR.